mixer/template/authz/template.proto

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package authz;
+
+import "mixer/v1/config/descriptor/value_type.proto";
+import "mixer/v1/template/extensions.proto";
+
+option (istio.mixer.v1.template.template_variety) = TEMPLATE_VARIETY_CHECK;
+
+// List of subjects
+message Subject {
+  // The user.
+  string user = 1;
+  // Groups the subject belongs to.
+  repeated string groups = 2;
+  // Additional attributes about the subject.
+  map<string, istio.mixer.v1.config.descriptor.ValueType> extra = 3;
```

Comment:
`extra` is not a great name, imho. can we use something like `attributes` or even something like `metadata` instead?  It might help to have a better comment on `Subject` that explains what a subject is (that doesn't use the word subject in the description) and what `extra` you might need/want for a subject.

Reply:
It was previously named "attributes". It was renamed because Martin thought that "attributes" conflicts with Istio terms of "attributes". It is not "metadata" either. This field represents the custom attributes (in addition to "user" and "groups") that can be used to identify the subject. It is hard to come up with a good name. Do you have other suggestions for the name?

Reply:
"properties"?